src/devices/things/ITemperatureSensor.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Iot.Device
```

Comment:
I am wondering if we should make this more Generic interface for the sensors report data (e.g. humidity, odometers...etc.) something like 


public interface ISensorData<T>
{
    T Data { get; }
}

Reply:
This is what @Ellerbach proposed as well. I generally like this idea but I think there are some issues with that.

See my response here:
- https://github.com/dotnet/iot/pull/256#issuecomment-465308671
- https://github.com/dotnet/iot/pull/256#issuecomment-465462304

the main thing being that some sensors have multiple embedded sensors (i.e. humidity & temeperature)

Reply:
> the main thing being that some sensors have multiple embedded sensors (i.e. humidity & temeperature)

Yeah, that would be a problem. :)